tests/ui/traits/negative-bounds/associated-constraints.rs

```diff
@@ -17,4 +16,7 @@ fn test3<T: !Trait<Assoc: Send>>() {}
 fn test4<T>() where T: !Trait<Assoc: Send> {}
 //~^ ERROR associated type constraints not allowed on negative bounds
 
+fn test5<T>() where T: !Fn() -> i32 {}
```

Comment:
This one used to ice with `no errors encountered even though `span_delayed_bug` issued` // `negative trait bounds should not have bindings`.